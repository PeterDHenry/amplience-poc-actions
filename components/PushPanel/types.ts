export interface PushPanelProps {
  height: number
  width: number
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  mainImg: { name: string }
  color: string
  fontWeight: number
  textAndCTA: {
    callToActionURL: string
    callToActionText: string
    headline: string
    strapline: string
  }
}

export interface PushPanelListProps {
  pushPanelList: PushPanelProps[]
  stackMobileLayout?: boolean
}
