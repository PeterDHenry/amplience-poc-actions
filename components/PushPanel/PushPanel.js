import React from 'react';
import styled from 'styled-components';
import PushPanelImage from '../LazyLoadImage/PushPanelImage';

const Panel = styled.div`
  max-width: 360px;
`

const PanelText = styled.div`
  width: 100%;
  min-height: 100px;
  padding: 10px;
  font-weight: 300;
  background: #FFF;
  font-family: "Roboto Condensed";
  padding-top: 10px;
  h2 {
    font-size: 32px;
  }
  h3 {
    font-size: 26px;
  }
  button {
    margin-top: 10px;
    border: 1.5px solid #000;
    padding: 4px 16px;
    font-size: 20px;
  }
`

const PushPanel = ( {color, fontWeight, content} ) => (

  <Panel>
  <a href={content.calltoactionurl}>
    <PushPanelImage id={content.background.name} />
    <PanelText>
      <h2 style={{color: color, fontWeight: fontWeight}}>
        {content.headline.toUpperCase()}
      </h2>
      <h3 style={{color: color}}>
        {content.strapline.toUpperCase()}
      </h3>
      <button style={{color: color}}>{content.calltoactiontext.toUpperCase()}</button>
    </PanelText>
    </a>
  </Panel>
);

PushPanel.defaultProps = {
  content: {
    background: {
      name: "pushpanel-1"
    },
    headline: "Latest Triathlon Kit",
    strapline: "Prepare for your next event",
    calltoactiontext: "Shop now",
    calltoactionurl: "https://www.wiggle.co.uk/triathlon-2021?range=ne"
  },
  color: "black"
}

export default PushPanel;
