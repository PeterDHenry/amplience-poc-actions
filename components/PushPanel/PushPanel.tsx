import React from 'react'
import styled from 'styled-components'
import PushPanelImage from '../LazyLoadImage/PushPanelImage'

const Panel = styled.div`
  max-width: 360px;
`

const PanelLink = styled.a`
  text-decoration: none;
`

const PanelText = styled.div`
  width: 100%;
  min-height: 100px;
  padding: 10px;
  font-weight: 300;
  background: #fff;
  font-family: 'Roboto Condensed';
  padding-top: 10px;
  h2 {
    font-size: 37px;
    min-height: 88px;
  }
  h3 {
    font-size: 25px;
    min-height: 60px;
  }
  button {
    margin-top: 10px;
    border: 1.5px solid #000;
    padding: 4px 16px;
    font-size: 20px;
  }
`

export interface PushPanelProps {
  _meta: {
    name: string
    schema: string
    deliveryId: string
  }
  mainImg: { name: string }
  calltoactiontext: string
  linkURL: string
  color: string
  fontWeight: number
  headerText: string
  subText: string
}

const PushPanel = ({
  mainImg,
  calltoactiontext,
  linkURL,
  color,
  fontWeight,
  headerText,
  subText,
}: PushPanelProps) => (
  <Panel>
    <PanelLink href={linkURL}>
      <PushPanelImage id={mainImg.name} />
      <PanelText>
        <h2 style={{ color: color, fontWeight: fontWeight }}>
          {headerText.toUpperCase()}
        </h2>
        <h3 style={{ color: color }}>{subText.toUpperCase()}</h3>
        <button style={{ color: color }}>
          {calltoactiontext.toUpperCase()}
        </button>
      </PanelText>
    </PanelLink>
  </Panel>
)

PushPanel.defaultProps = {
  mainImg: {
    name: 'homepageb3d2',
  },
  headerText: 'Test header text',
  subText: 'Test Sub text',
  calltoactiontext: 'Test CTA',
  linkURL: 'https://www.wiggle.co.uk/triathlon-2021?range=ne',
  color: 'black',
}

export default PushPanel
