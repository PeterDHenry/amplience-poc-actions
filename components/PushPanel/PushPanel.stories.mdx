import { Meta, Story, Canvas } from '@storybook/addon-docs/blocks'
import PushPanel from '../PushPanel/PushPanel'

<Meta
  title="CMS Content/PushPanel"
  component={PushPanel}
  argTypes={{
    color: {
      options: ['black', 'white', 'red', 'green', 'yellow', 'cyan'],
      control: { type: 'select' },
    },
    fontWeight: {
      options: ['light', 'normal', 'bold'],
      control: { type: 'select' },
    },
    mobileStack: {
      options: [true, false],
      control: { type: 'select' },
    },
  }}
  args={{
    color: 'black',
    fontWeight: 'light',
    mobileStack: true,
  }}
/>

export const Template = (args) => <PushPanel {...args} />

<Canvas>
  <Story name="Default">{Template.bind({})}</Story>
</Canvas>
