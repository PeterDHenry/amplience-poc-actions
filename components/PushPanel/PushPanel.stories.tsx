import { Meta, Story } from '@storybook/react'
import { PushPanel as Component } from './PushPanel'
import { defaultPanelProps, PushPanelProps } from '../PushPanel/types'

const meta: Meta = {
  title: 'CMS Content/PushPanel',
  component: Component,
  argTypes: {
    color: {
      options: ['black', 'white', 'red', 'green', 'yellow', 'cyan'],
      control: { type: 'select' },
    },
    fontWeight: {
      options: ['light', 'normal', 'bold'],
      control: { type: 'select' },
    },
    mobileStack: {
      options: [true, false],
      control: { type: 'select' },
    },
    cssClass: {
      options: ['col-lg-12', 'col-lg-6', 'col-lg-3'],
      control: { type: 'select' },
    },
  },
}

export default meta

type StoryProps = PushPanelProps

export const PushPanel: Story<StoryProps> = (props) => (
  <div className="container">
    <div className="row">
      <div className={`stack-${props.mobileStack} ${props.cssClass}`}>
        <Component {...props} />
      </div>
    </div>
  </div>
)

PushPanel.args = defaultPanelProps
