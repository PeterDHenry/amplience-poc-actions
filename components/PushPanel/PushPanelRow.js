import React from 'react';
import PushPanel from './PushPanel'

export default function CarouselsAll ({ pushPanelList }) {
    debugger
    return ( 
      <div className="container py-5">
        <div className="row">
          {( pushPanelList.map((panel)=>(
            <div className="col-xs-6 col-sm-4 w-pr-4 w-pl-2 w-pl-sm-4">
              <PushPanel  key={panel._meta.deliveryId}
                          {...panel}/>
            </div>
          )))}   
        </div>
      </div>
    )
};
  
