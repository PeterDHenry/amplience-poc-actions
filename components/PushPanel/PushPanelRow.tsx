import React from 'react'
import { PushPanel } from './PushPanel'
import { PushPanelProps, PushPanelRowProps, defaultPanelProps } from './types'

export const PushPanelRow = ({
  pushPanelList,
  stackMobileLayout,
}: PushPanelRowProps) => {
  let cssClass = `w-pr-4 w-pl-2 w-pl-sm-4 stack-${stackMobileLayout}`
  //let height = 500
  //let width = 500
  switch (pushPanelList.length) {
    case 2: {
      cssClass = 'col-xs-12 col-sm-6 ' + cssClass
      //height = 570
      //width = 795
      break
    }
    case 3: {
      cssClass = 'col-xs-6 col-sm-4 ' + cssClass
      break
    }
    case 4: {
      cssClass = 'col-xs-6 col-md-6 col-lg-3 ' + cssClass
      break
    }
  }
  return (
    <div className="container py-5">
      <div className="row">
        {pushPanelList.map((panel: PushPanelProps) => (
          <div
            key={panel._meta.deliveryId}
            className={`push-panel-item ${cssClass}`}
          >
            {/* <PushPanel {...panel} height={height} width={width} /> */}
            <PushPanel {...panel} />
          </div>
        ))}
      </div>
    </div>
  )
}

PushPanelRow.defaultProps = defaultPanelProps
