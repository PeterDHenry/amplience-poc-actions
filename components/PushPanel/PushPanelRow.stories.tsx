import { Meta, Story } from '@storybook/react'
import { PushPanelRow as Component } from './PushPanelRow'
import { defaultPanelRowProps, PushPanelRowProps } from './types'

const meta: Meta = {
  title: 'CMS Content/PushPanels',
  component: Component,
  argTypes: {
    color: {
      options: ['black', 'white', 'red', 'green', 'yellow', 'cyan'],
      control: { type: 'select' },
    },
    fontWeight: {
      options: ['light', 'normal', 'bold'],
      control: { type: 'select' },
    },
    mobileStack: {
      options: [true, false],
      control: { type: 'select' },
    },
  },
}

export default meta

type StoryProps = PushPanelRowProps

export const PushPanelRow: Story<StoryProps> = (props) => (
  <Component {...props} />
)

PushPanelRow.args = defaultPanelRowProps
