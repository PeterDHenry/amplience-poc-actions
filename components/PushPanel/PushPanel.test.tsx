import { render } from '@testing-library/react'
import PushPanel from './PushPanel'
import { PushPanelProps } from './types'

describe('PushPanel component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<PushPanel {...pushPanelProps} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})

const pushPanelProps: PushPanelProps = {
  height: 500,
  width: 500,
  _meta: {
    name: 'name',
    schema: 'schema',
    deliveryId: '328756873',
  },
  mainImg: {
    name: 'homepageb3d2',
  },
  textAndCTA: {
    callToActionURL: 'https://www.wiggle.co.uk/discover-different',
    headline: 'Default headline',
    strapline: 'Default Strapline',
    callToActionText: 'Default call to action',
  },
  color: 'black',
  fontWeight: 100,
}
