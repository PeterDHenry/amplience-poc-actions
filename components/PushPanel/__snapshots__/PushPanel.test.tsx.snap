// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PushPanel component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="PushPanel__Panel-sc-1magf3k-0 bTJEyg"
  >
    <a
      class="PushPanel__PanelLink-sc-1magf3k-1 gtiIsO"
      href="https://www.wiggle.co.uk/triathlon-2021?range=ne"
    >
      <div
        style="position: relative; display: block; padding-bottom: 33.85416666666667%;"
      >
        <noscript />
        <img
          alt="Prepare for your next event"
          data-src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepageb3d2?w=1920&h=650&sm=C"
          src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
          style="height: 100%; width: 100%; visibility: hidden; position: absolute; top: 0px; left: 0px;"
        />
        <div
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background: white; transition: 0.1s opacity;"
        />
      </div>
      <div
        class="PushPanel__PanelText-sc-1magf3k-2 gUZeSn"
      >
        <h2>
          Latest triathlon kit
        </h2>
        <h3>
          Prepare for your next event
        </h3>
        <button>
          Tri it out...
        </button>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
